Analysis/Tasks/PWGHF/HFCandidateCreator3Prong.cxx

```diff
@@ -170,18 +182,38 @@ struct HFCandidateCreator3ProngMC {
       // Λc± → p± K∓ π±
       if (result == N3ProngDecays) {
         //Printf("Checking Λc± → p± K∓ π±");
-        if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign) > -1) {
+        auto indexRecLc = RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign, -1);
+        if (indexRecLc > -1) {
           result = sign * LcToPKPi;
+
+          //Printf("Flagging the different Λc± → p± K∓ π± decay channels");
+          RecoDecay::getDaughters(particlesMC, indexRecLc, &arrDaughIndex, array{+kProton, -kKPlus, +kPiPlus}, 1);
```

Comment:
Since you ask for direct daughters, I don't see the point of providing the PDG codes of daughters, as the function will always return all the direct daughters from the first stage of recursion.